### ISSUE TITLE
AWS session-based SSO does not work with awskms encryption provider

### ISSUE THREAD

**Issue by midu-git** [pulumi/pulumi#14647]
### What happened?

Attempting to use AWS KMS as encryption provider with a profile using sso-session results in an error. pulumi login to an S3 backend while using AWS session-based SSO and get an error:

This seems to be exactly the issue as described in [this issue](https://github.com/pulumi/pulumi/issues/13142) and resolved by [this PR](https://github.com/pulumi/pulumi/pull/13619). See also [this comment](https://github.com/pulumi/pulumi/issues/13142#issuecomment-1649395944).

### Example

Configure your `~/.aws/config` as
```ini
[sso-session sso-session]                                                                                                                                                               
sso_start_url = https://d-1a2b3c4d5e6f.awsapps.com/start#                                                                                                                        
sso_region = eu-central-1                                                                                                                                                         
sso_registration_scopes = sso:account:access     

[profile profile-using-sso-session]
sso_session = sso-session
sso_account_id = 222222222222
sso_role_name = AdministratorAccess
region = eu-central-1
```

and perform, e.g., the following call
```shell
pulumi stack change-secrets-provider 'awskms://alias/your-kms-alias?region=eu-central-1&awssdk=v2&profile=profile-using-sso-session'
```

### Output of `pulumi about`

```shell
error: open keeper awskms://alias/your-kms-alias?region=eu-central-1&awssdk=v2&profile=profile-using-sso-session: profile "profile-using-sso-session" is configured to use SSO but is missing required configuration: sso_region, sso_start_url
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by justinvp** [comment]
@midu-git, does it work if you specify `awssdk=v3`, like:

```
pulumi stack change-secrets-provider 'awskms://alias/your-kms-alias?region=eu-central-1&awssdk=v3&profile=profile-using-sso-session'
```

**Comment by midu-git** [comment]
@justinvp Yes, this is the first step I've taken. When you try to change the encryption provider to use `v3` like
```shell
pulumi stack change-secrets-provider 'awskms://alias/your-kms-alias?region=eu-central-1&awssdk=v3&profile=profile-using-sso-session'
```
, the following output is generated
```shell
error: open keeper awskms://alias/your-kms-alias?region=eu-central-1&awssdk=v3&profile=profile-using-sso-session: unknown query parameter "profile"
```

**Comment by JonCholas** [comment]
Im facing this problem too, I want to move everyone to use the new session-based SSO and this is blocking us to move

**Comment by midu-git** [comment]
Hi @justinvp. Any rough estimate on when we can expect a solution?

**Comment by TheFynx** [comment]
@midu-git my PR only fixed the S3 Backend issue as it utilized the actual aws-sdk, KMS is using go.cloud, which was blocked from being updated (see https://github.com/pulumi/pulumi/blob/master/pkg/go.mod#L48C37-L49). 

It looks like the issue blocking that from being updated was resolved recently https://github.com/pulumi/pulumi/issues/11986 

So the solution is a PR that updates go.cloud to a version that has a newer AWS SDK supported that can do KMS with SSO Logins. 

I think all I did for my PR was update the go.mod, run the contributing checklist, then I built a binary, validated locally it worked, then submitted my PR. After that it's just waiting for Review and the Pulumi tests to be ran.
